tests/test_inputs.py

```diff
@@ -130,6 +156,69 @@ def test_samples(self):
         self.assertTrue(cqm.check_feasible(feasible_sample))
         self.assertFalse(cqm.check_feasible(infeasible_sample))
 
+    def test_states_nl(self):
+        shifts = [str(i + 1) for i in range(5)]
+
+        # Make every employee available for every shift
+        availability = {
+            "A-Mgr": [1] * 5,
+            "B-Mgr": [1] * 5,
+            "C": [1] * 5,
+            "D": [1] * 5,
+            "E": [1] * 5,
+            "E-Tr": [1] * 5,
+        }
+
+        test_params = utils.ModelParams(
+            availability=availability,
+            shifts=shifts,
+            min_shifts=1,
+            max_shifts=6,
+            shift_min=5,
+            shift_max=16,
+            requires_manager=True,
+            allow_isolated_days_off=False,
+            max_consecutive_shifts=6
+        )
+
+        model, assignments = employee_scheduling.build_nl(**asdict(test_params))
+
+        if not model.is_locked():
+            model.lock()
+
+        # Resize model states
+        model.states.resize(2)
+
+        feasible_state = [
+            [0, 0, 1, 1, 1], # A-Mgr
+            [1, 1, 0, 0, 0], # B-Mgr
+            [1, 1, 1, 1, 1], # C
+            [1, 1, 1, 1, 1], # D
+            [1, 1, 1, 1, 1], # E
+            [1, 1, 1, 1, 1]  # E-Tr
+        ]
+
+        # Infeasible: multiple managers scheduled for the same shift
+        infeasible_state = [
+            [1, 1, 1, 1, 1], # A-Mgr
+            [1, 1, 1, 1, 1], # B-Mgr
+            [1, 1, 1, 1, 1], # C
+            [1, 1, 1, 1, 1], # D
+            [1, 1, 1, 1, 1], # E
+            [1, 1, 1, 1, 1]  # E-Tr
+        ]
+
+        # Assign feasible state to index 0
+        assignments.set_state(0, feasible_state)
+        # Assign infeasible state to index 1
+        assignments.set_state(1, infeasible_state)
+
+        constraints_0 = [int(c.state(0)) for c in model.iter_constraints()]
+        constraints_1 = [int(c.state(1)) for c in model.iter_constraints()]
+
+        self.assertEqual(len(constraints_0), sum(constraints_0))
+        self.assertGreater(len(constraints_1), sum(constraints_1))
+
     def test_build_from_sample(self):
```

Comment:
I also noticed in the `test_build_from_sample` method that the availability specifies 14 shifts, but the sample only describes 5 shifts. Updated this in the `test_build_from_state` and can update that here as well.